packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,316 @@
+  fieldsToInclude?: string;
+  includeDefaultTemplate?: boolean;
+  onReady(): void;
+  onError(): void;
+}
+
+export class ResultListCommon {
+  private bindings: Bindings;
+  private render?: ResultRenderingFunction;
+  private updateBreakpoints?: (host: HTMLElement) => void;
+
+  public resultTemplatesManager!: ResultTemplatesManager<TemplateContent>;
+  public resultListControllerProps?: ResultListProps;
+
+  constructor(opts: ResultListCommonOptions) {
+    this.bindings = opts.bindings;
+    this.updateBreakpoints = once((host: HTMLElement) => {
+      updateBreakpoints(host);
+    });
+
+    if (opts.fieldsToInclude) {
+      this.resultListControllerProps = {
+        options: {
+          fieldsToInclude: this.determineAllFieldsToInclude(
+            opts.fieldsToInclude
+          ),
+        },
+      };
+    }
+
+    this.registerResultTemplates(
+      opts.templateElements,
+      opts.includeDefaultTemplate,
+      opts.onReady,
+      opts.onError
+    );
+  }
+
+  set renderingFunction(render: ResultRenderingFunction) {
+    this.render = render;
+  }
+
+  private determineAllFieldsToInclude(
+    configuredFieldsToInclude: string
+  ): string[] {
+    if (configuredFieldsToInclude.trim() === '')
+      return [...EcommerceDefaultFieldsToInclude];
+    return EcommerceDefaultFieldsToInclude.concat(
+      configuredFieldsToInclude.split(',').map((field) => field.trim())
+    );
+  }
+
+  public getTemplate(result: Result) {
+    return this.resultTemplatesManager.selectTemplate(result);
+  }
+
+  public getContentOfResultTemplate(
+    resultOrFolded: Result | FoldedResult
+  ): HTMLElement | DocumentFragment {
+    const result = (resultOrFolded as FoldedResult).result || resultOrFolded;
+    return this.render ? this.render(result) : this.getTemplate(result)!;
+  }
+
+  private makeDefaultTemplate(): ResultTemplate<DocumentFragment> {
+    const content = document.createDocumentFragment();
+    const linkEl = document.createElement('atomic-result-link');
+    content.appendChild(linkEl);
+    return {
+      content,
+      conditions: [],
+    };
+  }
+
+  public async registerResultTemplates(
+    elements: NodeListOf<TemplateElement>,
+    includeDefaultTemplate = true,
+    onReady: () => void,
+    onError: () => void
+  ) {
+    this.resultTemplatesManager = buildResultTemplatesManager(
+      this.bindings.engine
+    );
+
+    const customTemplates = await Promise.all(
+      Array.from(elements).map(async (resultTemplateElement) => {
+        const template = await resultTemplateElement.getTemplate();
+        if (!template) {
+          onError();
+        }
+        return template;
+      })
+    );
+    const templates = (
+      includeDefaultTemplate ? [this.makeDefaultTemplate()] : []
+    ).concat(
+      customTemplates.filter(identity) as ResultTemplate<DocumentFragment>[]
+    );
+
+    this.resultTemplatesManager.registerTemplates(...templates);
+    onReady();
+  }
+
+  public getResultId(
```

Comment:
if this isn't used outside of this component, I think it should be `private`.

Reply:
it is used outside